lib/streamlit/cursor.py

```diff
@@ -122,17 +184,22 @@ def __init__(self, path: CursorPath = (), index: int = 0, **props):
           for elements.
 
         """
+        self._container = container
         self._index = index
-        self._path = path
+        self._parent_path = parent_path
         self._props = props
 
     @property
-    def index(self) -> int:
-        return self._index
+    def container(self) -> Container:
```

Comment:
It's for sanity checking inside the DeltaGenerator constructor, and also so that the `Cursor` base class can construct the `delta_path` property. (We could avoid exposing this property by having the base class take it as a parameter, but I liked being able to avoid data inheritance here.)

Reply:
SG! If it were just the latter I would have wanted to make this private (prefixed with underscore), but the first use case seems reasonable.